src/test/lib/operators/import_test.cpp

```diff
@@ -18,55 +27,215 @@ class OperatorsImportTest : public BaseTest {
+    // For `.csv` and `.tbl` the default encoding should be `EncodingType::Dictionary`.
+    const auto encoding_spec = SegmentEncodingSpec{EncodingType::Dictionary};
+    ChunkEncoder::encode_all_chunks(expected_table, encoding_spec);
+  }
+
+  const auto reference_filename =
+      reference_filepath + reference_filenames.at(file_type) + file_extensions.at(file_type);
+  auto importer = std::make_shared<Import>(reference_filename, "a", Chunk::DEFAULT_SIZE, file_type, encoding);
   importer->execute();
+  auto table = Hyrise::get().storage_manager.get_table("a");
 
-  EXPECT_TABLE_EQ_ORDERED(Hyrise::get().storage_manager.get_table("a"), expected_table);
+  EXPECT_TABLE_EQ_ORDERED(table, expected_table);
+
+  auto segment = table->get_chunk(ChunkID{0})->get_segment(ColumnID{0});
+  auto expected_segment = expected_table->get_chunk(ChunkID{0})->get_segment(ColumnID{0});
+
+  // Check if the imported segment has the same type as the manually encoded one.
+  resolve_segment_type<float>(*segment, [&](const auto& segment) {
+    EXPECT_TRUE(std::dynamic_pointer_cast<std::remove_reference_t<decltype(segment)>>(expected_segment));
+  });
 }
 
-TEST_P(OperatorsImportMultiFileTypeTest, ImportWithoutFileType) {
+class OperatorsImportFileTypesTest : public OperatorsImportTest, public ::testing::WithParamInterface<FileType> {};
+
+std::string file_types_name_generator(const ::testing::TestParamInfo<FileType>& info) {
+  const auto& file_type = info.param;
+
+  return std::string{magic_enum::enum_name(file_type)};
+}
+
+INSTANTIATE_TEST_SUITE_P(FileTypes, OperatorsImportFileTypesTest,
+                         ::testing::Values(FileType::Csv, FileType::Tbl, FileType::Binary), file_types_name_generator);
+
+TEST_P(OperatorsImportFileTypesTest, ImportWithoutFileTypeAndEncoding) {
   auto expected_table =
       std::make_shared<Table>(TableColumnDefinitions{{"a", DataType::Float, false}}, TableType::Data, ChunkOffset{5});
   expected_table->append({1.1f});
   expected_table->append({2.2f});
   expected_table->append({3.3f});
   expected_table->append({4.4f});
 
+  // Make all chunks in the expected table immutable to allow encoding.
+  for (auto chunk_id = ChunkID{0}; chunk_id < expected_table->chunk_count(); ++chunk_id) {
+    expected_table->get_chunk(chunk_id)->set_immutable();
+  }
+
+  const auto& file_type = GetParam();
+  // For the `.csv` and `.tbl` files we expect a default encoding of `Dictionary`.
+  // The `.bin` file we load is stored and therefore expected to be `Unencoded`.
+  const auto encoding = (file_type == FileType::Binary) ? std::optional<EncodingType>{EncodingType::Unencoded}
+                                                        : std::optional<EncodingType>{EncodingType::Dictionary};
+
+  // Apply the encoding to the table.
+  if (encoding) {
+    const auto encoding_spec = SegmentEncodingSpec{*encoding};
+    ChunkEncoder::encode_all_chunks(expected_table, encoding_spec);
+  }
```

Comment:
You always set the enoding – why is it an optional. Plus, the table is unencoded already. So we can simplify to:
```suggestion
  if (file_type != FileType::Binary) {
    ChunkEncoder::encode_all_chunks(expected_table);
  }
```